solutions/main.tf

```diff
@@ -303,10 +303,117 @@ resource "aws_instance" "application" {
   vpc_security_group_ids = ["${aws_security_group.peekops_main_app.id}"]
   subnet_id = "${aws_subnet.peekops_main_app_a.id}"
   tags {
-     Name = "peekops_application"
+     Name = "peekops_application_a"
+  }
+  provisioner "local-exec" {
```

Comment:
I picked Terraform due to its similarity with puppet. But, it's a good chance for me to use other tools since I've never used them before.